AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/SubscriptionSync/AWSIncomingEventReconciliationQueueTests.swift

```diff
@@ -61,7 +61,7 @@ class AWSIncomingEventReconciliationQueueTests: XCTestCase {
             case .initialized:
                 expectInitialized.fulfill()
             default:
-                XCTFail("Should not expect any other state, received: \(event)")
+                break
```

Comment:
can you clarify why this and the other similar changes in this files do not consider this an unexpected state and fail anymore?

Reply:
This was poorly added since the CurrentValueSubject will now return the previously stored event and returning more than one event on the subscriber. I've updated the code to account for this, by fulfilling each expected event and failing on any other state (adding this XCTFail back)